### PR TITLE
Add validation for 2 positive integer fields

### PR DESCRIPTION
**Pull request by leemyongpakvn**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Field requires positive integer value need validation before saving
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#28929
| How to test?  | Check above issue

### CODE REVIEW

**Comment by okom3pom** [comment]
Hello @leemyongpakvn 

Thank you for your involvement in the project 
Could you try to use CS-fixer because the test will fail.

Code Before : 

Capture d’écran de 2022-07-02 12-32-53

Code After : 

Capture d’écran de 2022-07-02 12-33-14

Thank you again.

**Comment by leemyongpakvn** [comment]
@okom3pom Thanks for your tool suggestion. I have updated with php-cs-fixer's fixed code.

**Comment by leemyongpakvn** [comment]
We need a new trans string `'%s is invalid. Please enter an integer greater than %s.'` in `Admin.Notifications.Error `also.
Admin Notifications Error_GreaterThan

**Comment by leemyongpakvn** [comment]
It looks like there is problems with PHPStan itself, not the PR's code.
PHPStan_Error

File: productcomments.php [hunk 1]
```diff
@@ -164,13 +164,34 @@ public function getCacheId($id_product = null)
...
         if (Tools::isSubmit('submitModerate')) {
-            Configuration::updateValue('PRODUCT_COMMENTS_MODERATE', (int) Tools::getValue('PRODUCT_COMMENTS_MODERATE'));
-            Configuration::updateValue('PRODUCT_COMMENTS_ALLOW_GUESTS', (int) Tools::getValue('PRODUCT_COMMENTS_ALLOW_GUESTS'));
-            Configuration::updateValue('PRODUCT_COMMENTS_USEFULNESS', (int) Tools::getValue('PRODUCT_COMMENTS_USEFULNESS'));
-            Configuration::updateValue('PRODUCT_COMMENTS_COMMENTS_PER_PAGE', (int) Tools::getValue('PRODUCT_COMMENTS_COMMENTS_PER_PAGE'));
-            Configuration::updateValue('PRODUCT_COMMENTS_ANONYMISATION', (int) Tools::getValue('PRODUCT_COMMENTS_ANONYMISATION'));
-            Configuration::updateValue('PRODUCT_COMMENTS_MINIMAL_TIME', (int) Tools::getValue('PRODUCT_COMMENTS_MINIMAL_TIME'));
-            $this->_html .= '<div class="conf confirm alert alert-success">' . $this->trans('Settings updated', [], 'Modules.Productcomments.Admin') . '</div>';
+            $errors = [];
+            $pcmt = Tools::getValue('PRODUCT_COMMENTS_MINIMAL_TIME');
```
Comment (Amoifr) [inline productcomments.php:168]: acronym are banned in clean code best practice could you replace it ? 
```suggestion
            $productCommentsMinimalTime = Tools::getValue('PRODUCT_COMMENTS_MINIMAL_TIME');
```

File: productcomments.php [hunk 2]
```diff
@@ -164,13 +164,34 @@ public function getCacheId($id_product = null)
...
+            if (!Validate::isUnsignedId($pcmt)) {
```
Comment (Amoifr) [inline productcomments.php:169]: Why using isUnsignedId on a integer value that is not an id ?
```suggestion
            if (!Validate::isUnsignedInt($productCommentsMinimalTime)) {
```

**Review by Amoifr — COMMENTED** [review]
Thx for this PR. I report 2 types of improvement that can be made for my opinion. I let you take a look on these.

File: productcomments.php (hunk 1 above)
Comment (leemyongpakvn) [reply productcomments.php:168]: In fact, this acronym is explained inline. If you change variable name at line 168, then you need change it at line 191 also.
And make the same change for $pcpp at line 176 and 192 are needed for consistency.

**Review by kpodemski — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @leemyongpakvn ,

Thanks for the PR!

Tested the following on `1.7.8.x`, `8.0.x` and `develop` branch:
Special characters, accented characters, mix of integer/char, fractioned number, negative number.
If using any of the above characters, it won't save. I have an error message: ✅ 
Screenshot 2022-09-15 at 10 40 15

But if I enter `0` in BO, it will save. ❌ 
Screenshot 2022-09-15 at 10 40 26

Therefore, there's an error in FO, I can paginate infinitely: ❌ 

https://user-images.githubusercontent.com/16019289/190358554-c8de6c61-bdf7-4439-a90c-26cce556cb56.mov

Another thing is missing in the alert: the cross to close it. ❌ 
For example:
Screenshot 2022-09-15 at 10 27 30

Could you check ?
Thanks!

**Review by kpodemski — APPROVED** [review]

**Comment by leemyongpakvn** [comment]
@florine2623 Thanks for testing and informing. I have added one more positive number check and used Module::displayError for close button. @kpodemski has checked again and approved :)

**Review by florine2623 — APPROVED** [review]
Hello @leemyongpakvn ,

Tested on 1.7.8.x, 8.0.x, develop.
Looking great !

Alert message is fixed, 0 can't be saved, FO pagination is now OK.

It is QA ✅ !
Thanks!

**Comment by kpodemski** [comment]
Thanks @leemyongpakvn :)

**Pull request merged**